eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -324,25 +322,112 @@ function Invoke-Pack() {
     Pop-Location
 }
 function Get-ResourceProviderFromReadme($readmeFile) {
-    $readmeFileRegex = "(?<specName>.*)/(?<serviceType>.*)/readme.md"
-    $readmeFileRegexWithSpec = "specification/(?<specName>.*)/(?<serviceType>.*)/readme.md"
-    try
-    {
-        if ($readmeFile -match $readmeFileRegexWithSpec)
-        {
-            return $matches["specName"], $matches["serviceType"]
+    $readmeFile = $readmeFile -replace "\\", "/"
+    $pathArray = $readmeFile.Split("/");
+
+    if ( $pathArray.Count -lt 3) {
+        Write-Error "Error: invalid readme file path."
+        exit 1
+    }
+
+    $specName = $pathArray[-3]
+    $serviceType = $pathArray[-2]
+    Write-Host "specName:$specName, serviceType: $serviceType"
+
+    return $specName, $serviceType
+}
+
+function Invoke-GenerateAndBuildSDK () {
+    param(
+        [string]$readmeAbsolutePath,
+        [string]$sdkRootPath,
+        [object]$generatedSDKPackages
+    )
+    $readmeFile = $readmeAbsolutePath -replace "\\", "/"
+    Write-Host "readmeFile:$readmeFile"
+    $service, $serviceType = Get-ResourceProviderFromReadme $readmeFile
+    Write-Host "service:$service, serviceType:$serviceType"
+    
+    if (!(Test-Path -Path $readmeFile)) {
+        Write-Error "readme file does not exist."
+        exit 1
+    }
+    
+    $packagesToGen = @()
+    $newpackageoutput = "newPackageOutput.json"
+    if ( $serviceType -eq "resource-manager" ) {
+        Write-Host "Generate resource-manager SDK client library."
+        $package = $service
+        if ($packageNameHash[$service] -ne "") {
+            $package = $packageNameHash[$service]
         }
-        if ($readmeFile -match $readmeFileRegex)
-        {
-            return $matches["specName"], $matches["serviceType"]
+        New-MgmtPackageFolder -service $service -packageName $package -sdkPath $sdkRootPath -commitid $commitid -readme $readmeFile -outputJsonFile $newpackageoutput
+        if ( !$?) {
+            Write-Error "Failed to create sdk project folder. exit code: $?"
+            exit 1
+        }
+        $newpackageoutputJson = Get-Content $newpackageoutput | Out-String | ConvertFrom-Json
+        $packagesToGen = $packagesToGen + @($newpackageoutputJson)
+        Remove-Item $newpackageoutput
+    } else {
+        Write-Host "Generate data-plane SDK client library."
+        npx autorest --version=3.7.3 --csharp $readmeFile --csharp-sdks-folder=$sdkRootPath --skip-csproj --clear-output-folder=true
```

Comment:
Is there a reason we don't use the generate command? Such as https://github.com/Azure/azure-sdk-for-net/blob/main/eng/scripts/Automation-Sdk-Generate.ps1#L62? 

If we don't use that then we risk the generation process to get out of sync as we use that msbuild target in our builds that verify the generated code. We should share and use the same exact commands in both places.

Reply:
We need to know the package namespace(which will be the package folder name) at first, then to update/create the SDK package. Use this generation autorest generation command is an workaround to get the package folder name( or namespace), otherwise we need to parse readme.md ourself.

Reply:
Is this code path all about creating a new package? Do we need only the package name or the namespace (although I think they might be the same)? How did you avoid this for the mgmt case?

Reply:
@dw511214992 @chunyu3 I suspect this exact code path is what we are talking about in the email thread about where we should put the namespace and package name for new SDKs. Is there any way we could provide this information from the input file data?

Reply:
As we discussed in the meeting, the new SDKs will provide namespace and package name in the swagger PR comment, and it will be passed in the input file when call the script. This code path here is for existing SDKs, and the one which has more than one SDK packages one readme file, and it define 'batch' to generate multiple SDK packages. When all the RPs have convert to one readme one SDK package structure, we will not call 'autorest' cmd here.